tests/syntax_error/missing_self.jou

```diff
@@ -0,0 +1,8 @@
+class Foo:
+    def bar() -> void:
```

Comment:
Ideally the error message would be on this line. You can check it in the parser, maybe `parse_function_signature()`.